pkg/operator/controllers/cpms/cpms_controller.go

```diff
@@ -0,0 +1,100 @@
+package cpms
+
+import (
+	"context"
+
+	machinev1 "github.com/openshift/api/machine/v1"
+	"github.com/sirupsen/logrus"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	"github.com/Azure/ARO-RP/pkg/operator"
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+	"github.com/Azure/ARO-RP/pkg/operator/controllers/base"
+)
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+const (
+	ControllerName         = "CPMSDeactivatorController"
+	SingletonCPMSName      = "cluster"
+	SingletonCPMSNamespace = "openshift-machine-api"
+)
+
+type Reconciler struct {
+	base.AROController
+}
+
+func NewReconciler(log *logrus.Entry, client client.Client) *Reconciler {
+	return &Reconciler{
+		AROController: base.AROController{
+			Log:    log,
+			Client: client,
+			Name:   ControllerName,
+		},
+	}
+}
+
+// CPMS reconciler will disable the cluster CPMS if `aro.cpms.enabled` is false or missing.
```

Comment:
I think this comment is the perfect place in all this code to _not_ use `CPMS` and explicity state it as `Control Plane Machine Set` so we codify the meaning we are after very clearly without having to read the whole function and flow of the operator/controller.